met-web/src/services/membershipService/index.tsx

```diff
@@ -38,3 +38,30 @@ export const getMembershipsByUser = async ({
     const responseData = await http.GetRequest<EngagementTeamMember[]>(url);
     return responseData.data ?? [];
 };
+
+export const revokeMembership = async (engagement_id: number, membership_id: number): Promise<EngagementTeamMember> => {
+    const url = replaceAllInURL({
+        URL: Endpoints.EngagementTeamMembers.REVOKE,
+        params: {
+            engagement_id: String(engagement_id),
+            membership_id: String(membership_id),
+        },
+    });
+    const responseData = await http.PutRequest<EngagementTeamMember>(url);
+    return responseData.data;
+};
+
+export const reinstateMembership = async (
+    engagement_id: number,
+    membership_id: number,
+): Promise<EngagementTeamMember> => {
+    const url = replaceAllInURL({
+        URL: Endpoints.EngagementTeamMembers.REINSTATE,
+        params: {
+            engagement_id: String(engagement_id),
+            membership_id: String(membership_id),
+        },
+    });
+    const responseData = await http.PutRequest<EngagementTeamMember>(url);
```

Comment:
on another note , when we think about the revoking , do we generally revoke a membership from an engagement or user from engagement 

should the rest point accept , user_id and engaggement_id ? so thats its the whole users membership getting revoked instead of membership_id and engagemnet_id? I am not too sure about whats right.

Reply:
I am thinking we are revoking a membership, I thought about this. Either membership.id or engagement.id. The reason why I chose engagement.id is to for sure get the is_latest of the assignment for this user on this engagement and make it is_latest false. so I can add another membership with this engagement and user id. In my head user_id and engagement_id are like custom primary key

Reply:
yes if I fetch membership.id I get that membership. but with user_id and engagement_id I know I get the latest membership for that combined key

Reply:
Yes changed to user_id and engagement_id, I think we should treat that like custom combined primary key. I was thinking between the two and wasn't sure as well.